l10n_fr_department/model/res_partner.py

```diff
@@ -18,33 +20,35 @@ class ResPartner(models.Model):
     @api.depends("zip", "country_id", "country_id.code")
     # If a department code changes, it will have to be manually recomputed
     def _compute_department(self):
-        rcdo = self.env["res.country.department"]
+        def _get_zipcode(p) -> str:
+            if not p.country_id.id or p.country_id.id not in fr_country_ids:
+                return ""
+            p_zip = p.zip
+            if not p_zip or len(p_zip) != 5:
+                return ""
+            return p_zip.strip().replace(" ", "").rjust(5, "0")
```

Comment:
Seems weird to check for the len to be exactly `5` before the normalization done by  strip/rjust.
It would give false `""` in case of `p_zip = "    12345", no?

How about something like this instead?

```suggestion
            if not p_zip:
                return ""
            p_zip = p_zip.strip().replace(" ", "").rjust(5, "0")
            if not len(p_zip) != 5:
                return ""
            return p_zip
```

Reply:
@ivantodorovich seems weird to me too, but I wanted to keep this part as it was originally, without changing the business logic too much. Not sure what the initial idea was, so I don't feel confident in changing that...